src/main/java/seedu/address/model/room/Room.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model.room;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.UniqueGuestList;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a room in the hotel.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public abstract class Room {
+
+    // Identity fields
+    protected final RoomNumber roomNumber;
+
+    // Data fields
+    protected final Capacity capacity;
+    protected final UniqueGuestList occupants;
+    protected final Expenses expenses;
```

Comment:
Would a `Room` have more than one guest? Or are we only recording the mainGuest?

Reply:
Fair point. After doing some online research and making inquiries to Rendezvous Hotel (located at Bras Basah), I found out that the norm for hotel bookings is to keep track of just 1 guest per booking. Thus, each `Room` will only record **1 `Guest`**. 

Will change `UniqueGuestList occupants` to `Guest registeredGuest`.